### DISCUSSION TITLE
jsPsych 5 two tests in a row adds data from second test to first test data

### DISCUSSION THREAD

**Discussion by guttermouth** [jspsych/jsPsych#531] [Q&A]
Hello,

I am currently working with jsPsych 5 and ran into a problem.  I am running two tests in a row and for each test the data gets uploaded after it finishes (on_finish) to a cloud storage provider. The first test data uploads correctly however when the second test data gets uploaded the results from the first test and the second test merged somehow. I would like to clean the test data from the first test after continuing to the second test. Is there anyway to achieve this?  

Thanks for your help.

**Comment by jodeleeuw** [comment]
You can do the uploading inside the `on_data_update` callback. [Documentation here](https://github.com/jspsych/jsPsych/blob/v5.0.3/docs/markdown_docs/features/callbacks.md#on_data_update) for v5.0.3. If you send the contents of the argument passed to the callback then you'll only be storing the most recent data. The `jsPsych.data.getData()` array will still grow as the experiment runs.
